src/probnum/diffeq/_odesolver.py

```diff
@@ -158,25 +168,22 @@ def method_callback(self, state):
         current guess is accepted, but before storing it. No return. For
         example: tune hyperparameters (sigma).
         """
-        pass
-
 
-class _TimeStopper:
-    """Make the ODE solver stop at specified time-points."""
 
-    def __init__(self, locations: Iterable):
-        self._locations = iter(locations)
-        self._next_location = next(self._locations)
+def _time_stopper(locations):
+    """Check whether the next time-point is supposed to be stopped at."""
+    location_iter = iter(locations)
+    initial_location = next(location_iter)
 
-    def adjust_dt_to_time_stops(self, t, dt):
-        """Check whether the next time-point is supposed to be stopped at."""
-
-        if t >= self._next_location:
+    def adjust(t, dt, next_location=initial_location):
+        if t >= next_location:
             try:
-                self._next_location = next(self._locations)
+                next_location = next(location_iter)
             except StopIteration:
-                self._next_location = np.inf
+                next_location = np.inf
 
-        if t + dt > self._next_location:
-            dt = self._next_location - t
+        if t + dt > next_location:
+            dt = next_location - t
         return dt
+
+    return adjust
```

Comment:
I tried implementing this in a single function with partial(), but it did not work somehow. (At least not straightforwardly). But i think the present solution is also fine.